server/routes/workers.js

```diff
@@ -51,34 +20,38 @@ router.post('/', function (req, res, next) {
         trades: req.body.trades,
         pCode: req.body.pCode,
         imageURL: req.body.imageURL
-    };
-    workers.push(worker);
-    return res.send(worker);
+    });
+    await worker.save();
+    const workers = await Worker.find({})
+    return res.send(workers);
 });
-router.delete('/', function (req, res, next) {
+router.delete('/', async function (req, res, next) {
     const workerId = req.body.id;
-    workers = workers.filter(car => car.id !== workerId);
+    await Worker.deleteOne({id:workerId})
+    const workers = await Worker.find({});
     return res.send(workers);
 });
-router.put('/:userId', function(req, res, next){
-    const worker = { id: req.params.userId,
-        name: req.body.name,
-        email: req.body.email,
-        phone: req.body.phone,
-        address: req.body.address,
-        hRate: req.body.hRate,
-        trades: req.body.trades,
-        pCode: req.body.pCode,
-        imageURL: req.body.imageURL
-    };
-    const newWorkers = [...workers];
-    const index = workers.findIndex(workers => workers.id === req.params.userId);
-    newWorkers[index] = worker;
-    workers = newWorkers;
-    return res.send(worker);
+router.put('/:userId', async function(req, res, next){
+    try {
+    const foundWorker = await Worker.updateOne({ id: req.params.userId}, {$set: 
+        {
+            name: req.body.name,
+            email:req.body.email,
+            phone: req.body.phone,
+            address: req.body.address,
+            hRate: req.body.hRate,
+            trades: req.body.trades,
+            pCode: req.body.pCode,
+            imageURL: req.body.imageURL}});
+
+        res.status(StatusCodes.OK).json(foundWorker);
```

Comment:
repalce "res.status(StatusCodes.OK).json(foundWorker);" to "    const worker = await Worker.findOne({ id: req.params.userId})
        res.status(StatusCodes.OK).json(worker);
".
Otherwise, after click the update, it will not show the correct info.